client/network/src/protocol/sync.rs

```diff
@@ -1274,7 +1273,6 @@ impl<B: BlockT> ChainSync<B> {
 
 	/// Restart the sync process.
 	fn restart<'a>(&'a mut self) -> impl Iterator<Item = Result<(PeerId, BlockRequest<B>), BadPeer>> + 'a {
-		self.processed_blocks = 0;
```

Comment:
Why don't we reset `downloaded_blocks`?

Reply:
Total is more useful for tests and potentially for statistics.